osutil/syncdir.go

```diff
@@ -84,7 +90,17 @@ type MemoryFileState struct {
 
 // State returns a reader of the in-memory contents of a file, along with other meta-data.
 func (blob *MemoryFileState) State() (io.ReadCloser, int64, os.FileMode, error) {
-	return ioutil.NopCloser(bytes.NewReader(blob.Content)), int64(len(blob.Content)), blob.Mode.Perm(), nil
+	if err := checkFileType(blob.Mode); err != nil {
+		return nil, 0, os.FileMode(0), err
+	}
+	return ioutil.NopCloser(bytes.NewReader(blob.Content)), int64(len(blob.Content)), blob.Mode, nil
+}
+
+func checkFileType(mode os.FileMode) error {
```

Comment:
I don't think it makes sense to put this into a function. Without the context of the struct, it's not obvious why this is an error in itself. It also doesn't make the code more concise at the caller level

Reply:
updated, thanks for the review!